main.rb

```diff
@@ -0,0 +1,151 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index]) }
+    array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    array = is_a?(Range) ? to_a : self
+    array.length.times { |index| yield(array[index], index) }
+    array
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    array = []
+    my_each { |index| array.push(index) if yield(index) }
+    array
+  end
+
+  def my_all?(*arguments)
+    if !arguments[0].nil?
+      my_each { |index| return true unless arguments[0] == index }
+    elsif !block_given?
+      my_each { |index| return true unless index }
+    else
+      my_each { |index| return true unless yield(index) }
+    end
+    false
+  end
```

Comment:
# my_all? 
1. [ ] Returns true if the block never returns false or nil
### Test 👉 `expect(array.my_all?(&true_block)).to eq(array.all?(&true_block))`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|

##
2. [ ]  Returns false if the block returns false or nil
### Test 👉 `expect(array.my_all?(&false_block)).to eq(array.all?(&false_block))`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
3. [ ] Returns false if the block returns false or nil::range
### Test 👉 `expect(range.my_all?(&false_block)).to eq(range.all?(&false_block))`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
4. When no block or argument is given returns true when none of the collection members is false or nil
### Test 👉 `expect(true_array.my_all?).to be true_array.all?`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
##
5. [ ] When no block or argument is given returns false when one of the collection members is false or nil
### Test 👉 `expect(false_array.my_all?).to be false_array.all?`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
6. [ ] When a class is passed as an argument returns false if any of the collection is not a member of such class
### Test 👉 `expect(array.my_all?(Integer)).to be array.all?(Integer)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
7. [ ] When a Regex is passed as an argument returns false if any of the collection does not match the Regex
### Test 👉 `expect(words.my_all?(/o/)).to be words.all?(/o/)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
8. [ ] When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
### Test 👉 `expect(array.my_all?(3)).to be array.all?(3)`
 |Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
9. [ ] When a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
### Test  👉 `expect(array.my_all?(3)).to be array.all?(3)`
|Expected|Got|
|:---:|:---:|
|`false`|`true`|
##
10. [ ] Returns false if the block ever returns a value false or nil::range
### Test 👉 `expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))`
 |Expected|Got|
|:---:|:---:|
|`true`|`false`|
##
11. [ ]  When no block or argument is given returns true if at least one of the collection is not false or nil
### Test 👉 `expect(true_array.my_any?).to be true_array.any?`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
##

12. [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
## Test 👉 `expect(array.my_any?(Numeric)).to be array.any?(Numeric)` 
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
##
13. [ ] When a Regex is passed as an argument returns true if any of the collection matches the Regex
### Test 👉 `expect(words.my_any?(/d/)).to be words.any?(/d/)`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|
##
14. [ ]  When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
### Test 👉 `expect(words.my_any?('cat')).to be words.any?('cat')`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|